django_addanother/views.py

```diff
@@ -27,18 +31,23 @@ def form_valid(self, form):
             return response
 
     def respond_script(self, created_obj):
-        html = self.get_script_response_html(
-            escapejs(six.text_type(self._get_created_obj_pk(created_obj))),
-            escapejs(self.label_from_instance(created_obj)),
-        )
-        return http.HttpResponse(html, status=201)
-
-    def get_script_response_html(self, created_obj_pk, created_obj_label):
-        template = '<script>opener.dismissAddAnotherPopup(window, "{pk}", "{label}")</script>'
-        return template.format(
-            pk=created_obj_pk,
-            label=created_obj_label
-        )
+        if django.VERSION < (1, 10):
+            return SimpleTemplateResponse('admin/popup_response.html', {
+                'action': self.POPUP_ACTION,
+                'value': six.text_type(self._get_created_obj_pk(created_obj)),
+                'obj': six.text_type(self.label_from_instance(created_obj)),
+                'new_value': six.text_type(self._get_created_obj_pk(created_obj)),
+            })
+        else:
+            return SimpleTemplateResponse('admin/popup_response.html', {
+                'popup_response_data':
+                    '{"action":"%s","value":"%s","obj":"%s","new_value":"%s"}' % (
```

Comment:
Please use `json.dumps` here like in `django.contrib.admin`, which also saves you from using `escapejs`